AudiusProject/audius-protocol
# [AUD-531] Optimize CN nodesync saveFileForMultihash timeouts & concurrency

### Description
_What is the purpose of this PR? What is the current behavior? New behavior? Relevant links (e.g. Trello) and/or information pertaining to PR?_

- [x] query replicaset in parallel instead of sequential (with max concurrency)
- [x] move gateway retrieval to before ipfs ops to avoid the 2 sequential 1s timeouts
- [x] move all `fs` ops in module to async to prevent event loop blocking
- [x] change `saveFileForMultihash()` concurrency limit to node config to allow easy configuration

### Tests
_List the manual tests and repro instructions to verify that this PR works as anticipated. Include log analysis if possible.\
:exclamation: If this change impacts clients, make sure that you have tested the clients :exclamation:_

- all automated tests passing
- manual testing
  - required coverage:
    - `saveFileToIPFSFromFS` -> track upload
    - `saveFileFromBufferToIPFSAndDisk` -> user signup; track upload
    - `saveFileForMultihashToFS` -> nodesync
    - `removeTrackFolder` -> `fileManager.test.js`
  - user signup + track upload flow covers all `fileManager` changes + mad-dog covers sync

**:exclamation: Reminder :bulb::exclamation::**\
If this PR touches a critical flow (such as Indexing, Uploads, Gateway or the Filesystem), make sure to add the `requires-special-attention` label. **Add relevant labels as necessary.**
